src/AutoRest.CSharp/Mgmt/Output/MgmtExtensions.cs

```diff
@@ -41,7 +41,20 @@ private IEnumerable<MgmtClientOperation> EnsureClientOperations()
                     _context.Library.GetRestClient(operation),
                     operation.GetRequestPath(_context),
                     ContextualPath,
-                    GetOperationName(operation, ResourceName))));
+                    GetOperationName(operation, ResourceName),
+                    GetResourceFromResourceType(operation.GetRequestPath(_context).GetResourceType(_context.Configuration.MgmtConfiguration)))));
+        }
+
+        private Resource? GetResourceFromResourceType(ResourceTypeSegment resourceType)
+        {
+            var candidates = _context.Library.ArmResources.Where(resource => resource.ResourceType == resourceType);
+            if (candidates.Count() == 0)
+                return null;
+
+            if (candidates.Count() == 1)
+                return candidates.First();
+
+            throw new InvalidOperationException($"Found more than 1 candidate for {resourceType}, results were ({string.Join(',', candidates.Select(r => r.ResourceName))})");
```

Comment:
I have some concerns about this implementation here.
We have the case that multiple resources have the same resource type - see the MgmtExtensionResource for example.
This is a case for the same resource type staying under the different scopes (namely subscription, tenant and management group). In this test project, we do not have an operation in extension for them, therefore we did not get an exception during that project. And when the exception happens, we cannot resolve it.

Do we have a way to first escape this finding of candidates from a proper configuration? (a new or an existing one)

Reply:
I believe the check and throw we do will handle the case you describe.  Right now it has never happened but if it does we will need to adjust our implementation to handle it.  As long as this doesn't silently fail I am fine with it.